nap/utils.py

```diff
@@ -60,19 +51,21 @@ def flatten_params(k, vs):
 
         # since we can have more than one value for a single key, we use a
         # tuple of two tuples instead of a dictionary
-        params_tuple = tuple(itertools.chain(*flat_params))
-        param_string = urllib.parse.urlencode(params_tuple)
+        params_tuple = tuple(sorted(itertools.chain(*flat_params), key=lambda key_value: key_value[0]))
```

Comment:
Note: this was a pre-existing problem with normalize_url, but I figured I'd point it out now.
This isn't guaranteed to have the params in the same order if there are multi-value parameters present. We could add the value itself to be part of the key function.